prow/cmd/hook/main.go

```diff
@@ -223,42 +224,43 @@ func main() {
 	}
 
 	promMetrics := githubeventserver.NewMetrics()
-
-	defer interrupts.WaitForGracefulShutdown()
+	o.githubEventServerOptions.Metrics = promMetrics
 
 	// Expose prometheus metrics
 	metrics.ExposeMetrics("hook", configAgent.Config().PushGateway, o.instrumentationOptions.MetricsPort)
 	pjutil.ServePProf(o.instrumentationOptions.PProfPort)
 
 	server := &hook.Server{
-		ClientAgent:    clientAgent,
-		ConfigAgent:    configAgent,
-		Plugins:        pluginAgent,
-		Metrics:        promMetrics,
-		RepoEnabled:    o.githubEnablement.EnablementChecker(),
-		TokenGenerator: secretAgent.GetTokenGenerator(o.webhookSecretFile),
+		ClientAgent: clientAgent,
+		ConfigAgent: configAgent,
+		Plugins:     pluginAgent,
+		Metrics:     promMetrics,
+		RepoEnabled: o.githubEnablement.EnablementChecker(),
 	}
+
+	eventServer := githubeventserver.New(o.githubEventServerOptions, secretAgent.GetTokenGenerator(o.webhookSecretFile), logrus.NewEntry(logrus.New()))
+
+	eventServer.RegisterReviewEventHandler(server.HandleReviewEvent)
```

Comment:
The event server should just handle multiple plugins for one event, not needing the second-level server to do the multiplexing.

Reply:
This functionality is registered via the external plugins in [here](https://github.com/kubernetes/test-infra/pull/20717/files#diff-25f3adc74c147e59649ac9242dc3b9f462de7682b53810df1f8afe06485004fdR252)